build/tar.go

```diff
@@ -0,0 +1,80 @@
+package build
+
+import (
+	"archive/tar"
+	"io"
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+)
+
+func tarDir(destinationFilename, sourceDir string) {
+
+	if destinationFilename[len(destinationFilename)-3:] != "tar" {
+		log.Fatalln("Please provide a valid tar filename")
```

Comment:
We should not do `log.Fatalln` anywhere in `tar.go`. We should return the `error` and let the caller handle it properly.

Reply:
done